src/shogun/base/SGObject.h

```diff
@@ -100,6 +103,16 @@ template <class T> class SGStringList;
 
 #define SG_ADD(...) VARARG(SG_ADD, __VA_ARGS__)
 
+#define VALUE_TO_STRING_MACRO(s) #s
```

Comment:
long live the preprocessor :D

Reply:
it is so funny that this old school stuff still is necessary/useful in places

Reply:
yes unfortunately this is the simplest way..

Reply:
it seems like it might be possible to demangle it https://github.com/google/glog/blob/5c576f78c49b28d89b23fbb1fc80f54c879ec02e/src/demangle.cc#L1042, but would take a while to strip out the code we would need